bootstrap/internal/controllers/rke2config_controller.go

```diff
@@ -225,25 +240,329 @@ func (r *Rke2ConfigReconciler) SetupWithManager(mgr ctrl.Manager) error {
 // TODO: Implement these functions
 
 // handleClusterNotInitialized handles the first control plane node
-func (r *Rke2ConfigReconciler) handleClusterNotInitialized(ctx context.Context, scope *Scope) (res ctrl.Result, rerr error) {
-	return ctrl.Result{}, nil
-}
+func (r *RKE2ConfigReconciler) handleClusterNotInitialized(ctx context.Context, scope *Scope) (res ctrl.Result, reterr error) {
+
+	// initialize the DataSecretAvailableCondition if missing.
+	// this is required in order to avoid the condition's LastTransitionTime to flicker in case of errors surfacing
+	// using the DataSecretGeneratedFailedReason
+	// if conditions.GetReason(scope.Config, bootstrapv1.DataSecretAvailableCondition) != bootstrapv1.DataSecretGenerationFailedReason {
+	// 	conditions.MarkFalse(scope.Config, bootstrapv1.DataSecretAvailableCondition, clusterv1.WaitingForControlPlaneAvailableReason, clusterv1.ConditionSeverityInfo, "")
+	// }
+	if !scope.HasControlPlaneOwner {
+		scope.Logger.Info("Requeuing because this machine is not a Control Plane machine")
+		return ctrl.Result{RequeueAfter: 1 * time.Minute}, nil
+	}
+
+	if !r.RKE2InitLock.Lock(ctx, scope.Cluster, scope.Machine) {
+		scope.Logger.Info("A control plane is already being initialized, requeuing until control plane is ready")
+		return ctrl.Result{RequeueAfter: 1 * time.Minute}, nil
+	}
+
+	defer func() {
+		if reterr != nil {
+			if !r.RKE2InitLock.Unlock(ctx, scope.Cluster) {
+				reterr = kerrors.NewAggregate([]error{reterr, errors.New("failed to unlock the rke2 init lock")})
+			}
+		}
+	}()
+
+	certificates := secret.NewCertificatesForInitialControlPlane()
+	err := certificates.LookupOrGenerate(
+		ctx,
+		r.Client,
+		util.ObjectKey(scope.Cluster),
+		*metav1.NewControllerRef(scope.Config, bootstrapv1.GroupVersion.WithKind("RKE2Config")),
+	)
```

Comment:
```suggestion
	); err != nil {
```
nit: you can inline ifs in places where functions only return an error